language_detector/main.py

```diff
@@ -1,9 +1,53 @@
 """This is the entry point of the program."""
 
-from .languages import LANGUAGES
+try:
+    from .languages import LANGUAGES
+except:
+    from languages import LANGUAGES
 
 
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    spanish = []
+    english = []
+    german = []
+    #Grant easy access to the dictionaries
+    for lang in languages:
+        if lang["name"] == "Spanish":
+            spanish = lang["common_words"]
+        elif lang["name"] == "English":
+            english = lang["common_words"]
+        elif lang["name"] == "German":
+            german = lang["common_words"]
+            
+    #Create a variable with number of occurrences per language
+    english_occurrences = 0
+    german_occurrences = 0
+    spanish_occurrences = 0
+    
+    #Tokenize the text input
+    tokenized_text = text.split(" ")
+
+    
+    #count occurrences
+    #So we could iterate through that list
+    for word in tokenized_text:
+        #If the word is contained in the english common words list
+        if word in english:
+            #Increase the english occurrences
+            english_occurrences += 1
+            
+        elif word in spanish:
+            #increase the spanish occurences
+            spanish_occurrences += 1
+        elif word in german:
+            #increase the german occurences
+            german_occurrences += 1
+            
+    if english_occurrences > spanish_occurrences and english_occurrences > german_occurrences:
+        return "English"
+    elif spanish_occurrences > english_occurrences and spanish_occurrences > german_occurrences:
+        return "Spanish"
+    elif german_occurrences > english_occurrences and german_occurrences > spanish_occurrences:
+        return "German"
```

Comment:
As Jon has mentioned, harcoding languages again. Assuming you return some array or dictionary scoring the number of matches for each language, you could use some implementation of max(results) to find which language has the most matches.